lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,364 @@
+        params['replayId'] = nil
+        params['requireSignature'] = false
+        params['reviewIndicator'] = nil
+
+        params['risk'] = options['risk']
+
+        params['settledAmount'] = options['settledAmount']
+        params['settledCurrency'] = options['settledCurrency']
+        params['settledDate'] = options['created']
+        params['shipToCountry'] = options['shipToCountry']
+        params['shouldGetCreditCardLevel'] = options['shouldGetCreditCardLevel']
+        params['source'] = 'Spreedly'
+        params['sourceZip'] = nil
+        params['status'] = options['status']
+        params['tax'] = options['tax']
+        params['taxExempt'] = options['taxExempt']
+        params['tenderType'] = 'Card'
+        params['type'] = options['type']
+      end
+
+      def commit(action, params, iid, creditcardnumber, options)
+        response =
+          begin
+            if action == 'void' || action == 'close_batch'
+              ssl_invoke(action, params, iid, creditcardnumber, options)
+            else
+              parse(ssl_invoke(action, params, iid, creditcardnumber, options))
+            end
+          rescue ResponseError => e
+            parse(e.response.body)
+          end
+        success = success_from(response)
+
+        response = { 'code' => '204' } if response == ''
+        Response.new(
+          success,
+          message_from(success, response),
+          response,
+          authorization: success && response['code'] != '204' ? authorization_from(response) : nil,
+          error_code: success || response['code'] == '204' || response == '' ? nil : error_from(response),
+          test: test?
+        )
+      end
+
+      def handle_response(response)
+        if response.code != '204' && (200...300).cover?(response.code.to_i)
+          response.body
+        else
+
+          if response.code == '204' || response == ''
+            response.body = { 'code' => '204' }
+          else
+            raise ResponseError.new(response)
+          end
+        end
+      end
+
+      def ssl_invoke(action, params, refnumber, creditcardnumber, options)
+        if action == 'void'
+          ssl_request(:delete, url(action, params, refnumber, ''), nil, request_headers)
+        else
+          if action == 'verify'
+            ssl_get(url(action, params, '', creditcardnumber), request_verify_headers(options))
+          else
+            if action == 'get_payment_status' || action == 'create_jwt'
+              ssl_get(url(action, params, refnumber, ''), request_headers)
+            else
+              if action == 'close_batch'
+                ssl_request(:put, url(action, params, refnumber, ''), nil, request_headers)
+              else
+                ssl_post(url(action, params), post_data(params), request_headers)
+              end
+            end
+          end
+        end
+      end
+
+      def url(action, params, refnumber = '', creditcardnumber = '')
```

Comment:
This method is more traditionally used to determine whether or not to use a production, test, or other hostname for the given gateway.

Routes can and probably should be handled in the `commit` method where the `action` can be used to determine the `uri` and any values that need to be appended to that route.